public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -491,22 +491,26 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
         switchMap((res: { data: LokiStreamResponse; status: number }) =>
           iif(
             () => res.status === 404,
-            this._request(LEGACY_QUERY_ENDPOINT, target).pipe(
-              catchError((err: any) => {
-                const error: DataQueryError = {
-                  message: 'Error during context query. Please check JS console logs.',
-                  status: err.status,
-                  statusText: err.statusText,
-                };
-                throw error;
-              }),
-              map((res: { data: LokiLegacyStreamResponse }) => ({
-                data: res.data ? res.data.streams.map(stream => legacyLogStreamToDataFrame(stream, reverse)) : [],
-              }))
+            defer(() =>
```

Comment:
If I understand this correctly this is needed because the `_request` method creates a stream that is already hot (subscribed) and so request was already sent, right? So this will change it to cold one that will wait for actual subscription?